### PR TITLE
Support bundler local_search on 2.4.1 and higher

### LINKED ISSUE #596: Licensed breaks under Bundler 2.4.1 due to dependency search change
The latest version of bundler changed the local_search method in this commit - https://github.com/rubygems/rubygems/commit/7b64c64262a7a980c0eb23b96ea56cf72ea06e89 - to no longer accept a Gem::Dependency object and instead accept an array containing the gem name and version. That causes licensed to break here  - https://github.com/github/licensed/blob/a6272b166d9dbfac2c6c7ec3d7d443ba17504be2/lib/licensed/sources/bundler/definition.rb#L21

I have a fix locally that I'll push as a PR momentarily but I wanted to log this as an issue in case that is the preferred way to approach a change on this project.

### PR DESCRIPTION
**Pull request by JoshReedSchramm**
**Description**

This updates the call to local_search in bundler for the new query format on version 2.4.1 or higher. 

Fixes #596

### CODE REVIEW

**Comment by JoshReedSchramm** [comment]
A couple notes

- I tried to maintain support for old and new versions of bundler but if there's a better way to do the version comparison I'd love the feedback. 
- Given that this is for the version of bundler itself I wasn't sure how to add a test to the bundler tests for his scenario. 
- Also I ran into issues trying to run tests locally even before I made this change, even after following the steps in the contributor guide.

**Review by jonabc — COMMENTED** [review]
thank you so much for opening this PR! I left two suggested changes, with the suggestion for changing the version check being more critical than the other.

As a small style nit, because the change in logic between the two versions is so limited, it would be nice to have the logic and version check encapsulated to a separate method.  If you'd prefer not to handle that in this PR that's totally fine, I can update it as a follow up.  I'm thinking of something like

```ruby
def bundler_query
  if Gem::Version.new(::Bundler::VERSION) >= Gem::Version.new("2.4.0")
    ["bundler", ::Bundler.gem_version]
  else
    Gem::Dependency.new("bundler", ::Bundler::VERSION)
  end
end
```

**Comment by jonabc** [comment]
> - Given that this is for the version of bundler itself I wasn't sure how to add a test to the bundler tests for his scenario.

The way to test this is to add '2.4' to the [list of supported bundler versions](https://github.com/github/licensed/blob/a6272b166d9dbfac2c6c7ec3d7d443ba17504be2/.github/workflows/test.yml#L31) in CI .  If that passes with the changes in this PR then all good!

> - Also I ran into issues trying to run tests locally even before I made this change, even after following the steps in the contributor guide.

Can you describe what issues you ran into, assuming they're related to repository setup and not specific to using bundler >= 2.4.0 which is fixed by this PR.

**Comment by jonabc** [comment]
oh and as one final comment, don't worry about the failing license metadata CI check.  It's an issue with using a PAT in a PR opened from a fork.  I still haven't figured out a good way to get that working and it's not a required check 🤷

File: lib/licensed/sources/bundler/definition.rb [hunk 1]
```diff
@@ -18,7 +18,12 @@ def specs
 
           all_dependencies = requested_dependencies.concat(specs.flat_map(&:dependencies))
           if all_dependencies.any? { |d| d.name == "bundler" } && !specs["bundler"].any?
-            bundler = sources.metadata_source.specs.search(Gem::Dependency.new("bundler", ::Bundler::VERSION)).last
+            if ::Bundler::VERSION >= "2.4.1"
```
[the remark this answers is not in the document]
Comment (JoshReedSchramm) [reply lib/licensed/sources/bundler/definition.rb:21]: I had a feeling the string check was the wrong approach but didn't know about Gem::Version comparison - that's perfect thanks.

**Comment by jonabc** [comment]
🚀 thanks so much for fixing this. FYI the next release, which will include this fix, will bump the major version due to containing some breaking changes in the release process outside of my control.

**Review by jonabc — APPROVED** [review]

**Pull request merged**